rotkehlchen/data_migrations/migrations/migration_11.py

```diff
@@ -42,10 +47,45 @@ def _get_optimism_transaction_fees(rotki: 'Rotkehlchen', progress_handler: 'Migr
         )
 
 
+def _detect_arbitrum_one_accounts(rotki: 'Rotkehlchen', progress_handler: 'MigrationProgressHandler') -> None:  # noqa: E501
+    """
+    Detect accounts that have activity in other evm chains and are not yet tracked.
+    The reason is to start the new release with arbitrum one already detected.
+
+    Initially we check updates for spam assets and rpc nodes. Spam assets are updated before
+    accounts detection to avoid detecting accounts that only have spam assets. We also check
+    for rpc nodes updates here, to avoid an edge case where a call to rpc nodes happens before the
+    rpc node update task has finished.
+    """
+    progress_handler.new_step('Fetching new spam assets and rpc data info')
+    rotki.data_updater.check_for_updates(updates=[UpdateType.SPAM_ASSETS, UpdateType.RPC_NODES])
+
+    with rotki.data.db.conn.write_ctx() as write_cursor:
+        rotki.data.db.add_external_service_credentials(
+            write_cursor=write_cursor,  # add temporary etherscan arbitrum one key
+            credentials=[ExternalServiceApiCredentials(
+                service=ExternalService.ARBITRUM_ONE_ETHERSCAN,
+                api_key=ApiKey('VQUFYKKJR4RK8HFHYIJ9I93FIUVP44TN99'),  # same one in tests
+            )])
+
+    rotki.chains_aggregator.detect_evm_accounts(progress_handler)
+    # remove temporary etherscan arbitrum one key
+    rotki.data.db.delete_external_service_credentials([ExternalService.ARBITRUM_ONE_ETHERSCAN])
+
+
 def data_migration_11(rotki: 'Rotkehlchen', progress_handler: 'MigrationProgressHandler') -> None:
     """
     Introduced at v1.30.0
     """
     log.debug('Enter data_migration_11')
-    _get_optimism_transaction_fees(rotki, progress_handler)
+    with rotki.data.db.conn.read_ctx() as cursor:
+        # We need to get the optimism transactions here in order to calculate the number of
+        # steps for the migration.
+        cursor.execute('SELECT tx_hash, identifier FROM evm_transactions WHERE chain_id=10')
+        txhash_to_id = dict(cursor)
+
+    # steps are: optimism txs + ethereum accounts + potentially write to db + updating spam assets and rpc nodes + step 0  # noqa: E501
```

Comment:
What do you mean by step 0 here? What's the +3?

1. updating spam assets + rpc nodes
2. ?
3. ?

Reply:
Step 0 is not the best name. It is the message for the new migration round. I changed the name in the comment to be more self-explanatory
```
    def _perform_migration(self, migration: MigrationRecord) -> bool:
        self.progress_handler.new_round(version=migration.version)
        try:
            migration.function(self.rotki, self.progress_handler)
```
Also updated the comment for what 3 means:
1. potentially write to db
2. updating spam assets and rpc nodes
3. new round message